File: leetcode/python/twosum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        complements: dict[int, int] = {}

        for i, num in enumerate(nums):
            complement = target - num

            if complement in complements:
                return [complements[complement], i]
            else:
                complements[num] = i

        return []

File: leetcode/java/twosum.java
package leetcode.java;

import java.util.HashMap;

public class twosum {
  public int[] twoSum(int[] nums, int target) {
    var hm = new HashMap<Integer, Integer>();

    for (int i = 0; i < nums.length; ++i) {
      int num = nums[i];
      int complement = target - num;

      if (hm.containsKey(complement)) return new int[] {hm.get(complement), i};

      hm.put(num, i);
    }

    return new int[] {};
  }
}
